instrumentation/runtime/doc.go

```diff
@@ -12,21 +12,80 @@
-//   runtime.go.mem.heap_released (bytes)    Bytes of idle spans whose physical memory has been returned to the OS
-//   runtime.go.mem.heap_sys      (bytes)    Bytes of heap memory obtained from the OS
-//   runtime.go.mem.live_objects  -          Number of live objects is the number of cumulative Mallocs - Frees
-//   runtime.uptime               (ms)       Milliseconds since application was initialized
+//   process.runtime.go.cgo.calls         -          Number of cgo calls made by the current process
+//   process.runtime.go.gc.count          -          Number of completed garbage collection cycles
+//   process.runtime.go.gc.pause_ns       (ns)       Amount of nanoseconds in GC stop-the-world pauses
+//   process.runtime.go.gc.pause_total_ns (ns)       Cumulative nanoseconds in GC stop-the-world pauses since the program started
+//   process.runtime.go.goroutines        -          Number of goroutines that currently exist
+//   process.runtime.go.lookups           -          Number of pointer lookups performed by the runtime
+//   process.runtime.go.mem.heap_alloc    (bytes)    Bytes of allocated heap objects
+//   process.runtime.go.mem.heap_idle     (bytes)    Bytes in idle (unused) spans
+//   process.runtime.go.mem.heap_inuse    (bytes)    Bytes in in-use spans
+//   process.runtime.go.mem.heap_objects  -          Number of allocated heap objects
+//   process.runtime.go.mem.heap_released (bytes)    Bytes of idle spans whose physical memory has been returned to the OS
+//   process.runtime.go.mem.heap_sys      (bytes)    Bytes of heap memory obtained from the OS
+//   process.runtime.go.mem.live_objects  -          Number of live objects is the number of cumulative Mallocs - Frees
+//   runtime.uptime                       (ms)       Milliseconds since application was initialized
+//
+// The Go-1.16 release featured a new runtime/metrics package that gives formal
+// metric names to the various quantities.  This package supports the new metrics
+// under their Go-specified names by setting `WithUseGoRuntimeMetrics(true)`.
+// These metrics are documented at https://pkg.go.dev/runtime/metrics#hdr-Supported_metrics.
+//
+// The `runtime/metrics` implementation will replace the older
+// implementation as the default no sooner than January 2023. The
+// older implementation will be removed no sooner than January 2024.
+//
+// The following metrics are generated in go-1.19.
+//
+// Name                                                    Unit         Instrument
+// -----------------------------------------------------------------------------
+// process.runtime.go.cgo.go-to-c-calls                    {calls}      Counter[int64]
+// process.runtime.go.gc.cycles.automatic                  {gc-cycles}  Counter[int64]
+// process.runtime.go.gc.cycles.forced                     {gc-cycles}  Counter[int64]
+// process.runtime.go.gc.cycles.total                      {gc-cycles}  Counter[int64]
```

Comment:
Should we omit total from the name (since it is implied?)

Reply:
This raises more issues, unfortunately, and we can't separate this from the ongoing debate about "direction" attributes. I studied these two metrics with a `.total`, and it turns out they are duplicative. For every `.total` metric there are a number of other metrics that sum to the total.

This leaves options: 
1. Drop the total metric? Declare that quantities like gc.cycles.forced and gc.cycles.automatic are separate?
2. Drop the total metric? Merge all the things it totals into the same metric with an attribute for each?

Reply:
For the record, I prefer to drop the total and combine all sub-components into a single metric. This will take some work.

Reply:
This is now done. There are 13 timeseries with name `process.runtime.go.memory.classes` and `class=...`, there are 2 timeseries with name `process.runtime.go.gc.cycles` and `cycle=automatic` or `cycle=forced`.